docker/cuda/Dockerfile

```diff
@@ -14,6 +14,13 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     echo "deb https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64 /" > /etc/apt/sources.list.d/nvidia-ml.list && \
     apt-key adv --fetch-keys http://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/7fa2af80.pub && \
     apt-key adv --fetch-keys http://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub && \
+    if [[ -z "${ADD_2204_APT_REPOS}" ]] ; \
```

Comment:
is it necessary to make this conditional?

Reply:
@rokroskar yes that is needed. Without it the older libraries for older cuda versions simply cannot be found it seems:

```
> [3/7] RUN apt-get update && apt-get install -y --no-install-recommends     cuda-cudart-11-5=11.5.117-1     cuda-compat-11-5     && ln -s cuda-11.5 /usr/local/cuda &&     rm -rf /var/lib/apt/lists/*:
0.595 Get:18 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [25.6 kB]
0.596 Get:19 http://archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [49.2 kB]
0.606 Get:20 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [833 kB]
0.663 Get:21 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [44.0 kB]
0.664 Get:22 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [848 kB]
0.672 Get:23 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [979 kB]
0.740 Get:24 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64  Packages [2,445 B]

3.751 E: Unable to locate package cuda-cudart-11-5
3.751 E: Unable to locate package cuda-compat-11-5
------
Dockerfile:30
--------------------
  29 |     # For libraries in the cuda-compat-* package: https://docs.nvidia.com/cuda/eula/index.html#attachment-a
  30 | >>> RUN apt-get update && apt-get install -y --no-install-recommends \
  31 | >>>     $CUDA_CUDART_PACKAGE \
  32 | >>>     $CUDA_COMPAT_PACKAGE \
  33 | >>>     && ln -s cuda-${CUDA_VERSION} /usr/local/cuda && \
  34 | >>>     rm -rf /var/lib/apt/lists/*
  35 |     
--------------------
ERROR: failed to solve: process "/bin/bash -c -o pipefail apt-get update && apt-get install -y --no-install-recommends     $CUDA_CUDART_PACKAGE     $CUDA_COMPAT_PACKAGE     && ln -s cuda-${CUDA_VERSION} /usr/local/cuda &&     rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
```

See: https://github.com/SwissDataScienceCenter/renkulab-docker/actions/runs/5753730715/job/15597717590?pr=368